src/main/java/upb/dice/twitter/TweetExtractor.java

```diff
@@ -2,55 +2,81 @@
  */
 public class TweetExtractor {
+
     private StoreMaxID storeMaxID = new StoreMaxID();
     private Twitter twitter = TwitterFactory.getSingleton();
-    private RateLimitChecker rateLimitChecker = new RateLimitChecker();
-    private boolean check;
+    private boolean remainingTweet;
+    public boolean rateLimit;
 
     /**
      * Recursive search for tweets based on this query and store the maxID for the search results
      *
      * @param query search based on this query
      */
-    public void getTweet(Query query) throws IOException, TwitterException {
+    public void getTweet(Query query) throws InterruptedException {
         QueryResult queryResult;
-        check = rateLimitChecker.rateLimitCheck();
-        System.out.println("Tweets search Start");
-        FileWriter writer = new FileWriter("Tweets.txt", true);
-        int counter = 0;
         try {
+            rateLimit = new RateLimitChecker().rateLimitCheck();
+            System.out.println("Tweets search Start");
+            String directoryName = "Tweets Search Details";
+            File file = new File("Tweets Search Details" + File.separator + "Tweet object.txt");
+            FileWriter writer = new FileWriter(file, true);
+            int counter = 0;
+            long sinceID = 0;
+            long oldestTweetID;
             do {
-                check = rateLimitChecker.rateLimitCheck();
                 queryResult = twitter.search(query);
                 List<Status> queryTweets = queryResult.getTweets();
-                long maxId;
-                if (counter == 0) {
-                    maxId = queryResult.getMaxId();
-                    if (query.getQuery() == null) {
-                        storeMaxID.locationBasedMaxID(maxId, query);
-                    } else {
+                long maxId = queryResult.getMaxId();
+
+                //the first page and the first time
+                if (counter == 0 && !remainingTweet) {
+                    sinceID = queryResult.getSinceId();
+                    if (query.getQuery() != null) {
                         storeMaxID.keywordBasedMaxID(maxId, query);
                     }
+                    else {
+                        storeMaxID.locationBasedMaxID(maxId, query);
+                    }
                 }
-                counter++;
+                oldestTweetID = 0;
                 for (Status i : queryTweets) {
+                    oldestTweetID = i.getId();
                     writer.append(i.toString());
                 }
+                counter++;
+                rateLimit = new RateLimitChecker().rateLimitCheck();
+            }
+            while (((query = queryResult.nextQuery()) != null) && (!rateLimit) && oldestTweetID > sinceID);
+
+            //if the rate limit has reached and there is a gap between the base and the current tweet, redo the search with the oldest tweet as the upper bound
+            if (oldestTweetID - sinceID > 0 && rateLimit && query != null) {
```

Comment:
where are the `oldestTweetID` and the `sinceID` stored? I can only see the storing of the `maxId`.

Reply:
oldestTweetID and sinceID storage was circumvented